src/condor_tools/history.cpp

```diff
@@ -415,23 +417,26 @@ main(int argc, const char* argv[])
 		}
 	}
 	else if (is_dash_arg_colon_prefix(argv[i], "epoch", &pcolon, 1)) { //TODO: Add flag to usage when ready to share with the world
+		//Reset to NULL if previously set else where
+		searchDirectory.clear();
+		matchFileName.clear();
 		//Designate reading job run instance ads from epoch directory
 		read_epoch_ads = true;
 		//Get aggregate epoch history file
-		epochHistoryFile = param("JOB_EPOCH_HISTORY");
+		matchFileName.set(param("JOB_EPOCH_HISTORY"));
 		//If epoch file is NULL then check for an epoch directory
-		if (epochHistoryFile == NULL) {
+		if (!matchFileName) {
 			//Get epoch directory and validate passed arg is a directory
-			epochDirectory = param("JOB_EPOCH_INSTANCE_DIR");
-			if ( epochDirectory != NULL ) {
-				StatInfo si(epochDirectory);
-				if (!si.IsDirectory() ) {
-					fprintf(stderr, "Error: %s is not a valid directory.\n", epochDirectory);
+			searchDirectory.set(param("JOB_EPOCH_INSTANCE_DIR"));
+			if (searchDirectory) {
+				StatInfo si(searchDirectory.ptr());
```

Comment:
.ptr() needed here?

Reply:
It appears so. I tried to do just searchDirectory and got a compilation error stating: `error: call to constructor of 'StatInfo' is ambiguous`